Compile error moving from 3.5.9 to 4.0.8

### Version of react-native-iap
"react-native-iap": "^4.0.8",

### Version of react-native
react-native-cli: 2.0.1
react-native: 0.60.5

### Platforms you faced the error (IOS or Android or both?)
Android

### Expected behavior
react-native run-android to result in a successful build

### Actual behavior
react-native run-android errors with the error:


```
$ react-native run-android
error React Native CLI uses autolinking for native dependencies, but the following modules are linked manually: 
  - react-native-vector-icons (to unlink run: "react-native unlink react-native-vector-icons")
This is likely happening when upgrading React Native from below 0.60 to 0.60 or above. Going forward, you can unlink this dependency via "react-native unlink <dependency>" and it will be included in your app automatically. If a library isn't compatible with autolinking, disregard this message and notify the library maintainers.
Read more about autolinking: https://github.com/react-native-community/cli/blob/master/docs/autolinking.md
info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
Jetifier found 1160 file(s) to forward-jetify. Using 8 workers...
info JS server already running.
info Installing the app...

> Task :react-native-iap:compileDebugJavaWithJavac FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.4.1/userguide/command_line_interface.html#sec:command_line_warnings
42 actionable tasks: 2 executed, 40 up-to-date
warning: [options] source value 7 is obsolete and will be removed in a future release
warning: [options] target value 7 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
/home/fabian/Projects/poppy-app/node_modules/react-native-iap/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java:525: error: cannot find symbol
        promiseItem = item.copy();
                          ^
  symbol:   method copy()
  location: variable item of type WritableMap
Note: /home/fabian/Projects/poppy-app/node_modules/react-native-iap/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error
3 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-iap:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s

error Failed to install the app. Make sure you have the Android development environment set up: https://facebook.github.io/react-native/docs/getting-started.html#android-development-environment. Run CLI with --verbose flag for more details.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
warning: [options] source value 7 is obsolete and will be removed in a future release
warning: [options] target value 7 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
/home/fabian/Projects/poppy-app/node_modules/react-native-iap/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java:525: error: cannot find symbol
        promiseItem = item.copy();
                          ^
  symbol:   method copy()
  location: variable item of type WritableMap
Note: /home/fabian/Projects/poppy-app/node_modules/react-native-iap/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error
3 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-iap:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s

    at checkExecSyncError (child_process.js:621:11)
    at execFileSync (child_process.js:639:15)
    at runOnAllDevices (/home/fabian/Projects/poppy-app/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:74:39)
    at buildAndRun (/home/fabian/Projects/poppy-app/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/index.js:158:41)
    at /home/fabian/Projects/poppy-app/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/index.js:125:12
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Command.handleAction (/home/fabian/Projects/poppy-app/node_modules/react-native/node_modules/@react-native-community/cli/build/cliEntry.js:160:7)
```



### Tested environment (Emulator? Real Device?)
Real device

### Steps to reproduce the behavior
Upgrade from 3.5.9 to 4.0.8

--- a/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java
+++ b/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java
@@ -522,7 +522,8 @@ public class RNIapModule extends ReactContextBaseJavaModule implements Purchases
         item.putBoolean("isAcknowledgedAndroid", purchase.isAcknowledged());
         item.putInt("purchaseStateAndroid", purchase.getPurchaseState());
 
-        promiseItem = item.copy();
+        promiseItem = new WritableNativeMap();
+        promiseItem.merge(item);
         sendEvent(reactContext, "purchase-updated", item);
       }
       if (purchases.size() > 0 && promiseItem != null) {

